relay/messages/message.go

```diff
@@ -55,38 +59,14 @@ type HelloResponse struct {
 	InstanceAddress string
 }
 
-// DetermineClientMsgType determines the message type from the first byte of the message
-func DetermineClientMsgType(msg []byte) (MsgType, error) {
-	msgType := MsgType(msg[0])
-	switch msgType {
-	case MsgTypeHello:
-		return msgType, nil
-	case MsgTypeTransport:
-		return msgType, nil
-	case MsgTypeClose:
-		return msgType, nil
-	case MsgTypeHealthCheck:
-		return msgType, nil
-	default:
-		return 0, fmt.Errorf("invalid msg type, len: %d", len(msg))
-	}
-}
-
-// DetermineServerMsgType determines the message type from the first byte of the message
-func DetermineServerMsgType(msg []byte) (MsgType, error) {
-	msgType := MsgType(msg[0])
-	switch msgType {
-	case MsgTypeHelloResponse:
-		return msgType, nil
-	case MsgTypeTransport:
-		return msgType, nil
-	case MsgTypeClose:
-		return msgType, nil
-	case MsgTypeHealthCheck:
-		return msgType, nil
-	default:
-		return 0, fmt.Errorf("invalid msg type (len: %d)", len(msg))
+// DetermineMessageType determines the message type and version from the first two bytes of the message
+func DetermineMessageType(msg []byte) (byte, MsgType, error) {
+	if len(msg) < 2 {
+		return 0, 0, fmt.Errorf("invalid message length")
 	}
+	version := msg[0]
+	msgType := MsgType(msg[1])
+	return version, msgType, nil
```

Comment:
can you share why we don't need the message type validation here?

Reply:
That was a mistake, fixed